editor/scene_tree_dock.cpp

```diff
@@ -1024,17 +1029,19 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 					new_scene_from_dialog->set_title(TTR("Save New Scene As..."));
 					new_scene_from_dialog->popup_file_dialog();
 				} else {
-					if (da->file_exists(current_dir + existing)) {
+					String dir = new_scene_from_dialog->get_current_dir().path_join(existing);
```

Comment:
It would be better to store `new_scene_from_dialog->get_current_dir()` in a variable outside the loop, to not call it for each node.